src/services/EcoBridge/Socket/SocketBridge.ts

```diff
@@ -468,42 +402,214 @@ export class SocketBridge extends EcoBridgeChildBase {
+
+    // reset previous data
+    this.store.dispatch(
+      this.actions.setApprovalData({
+        allowanceTarget: undefined,
+        amount: undefined,
+        chainId: undefined,
+        owner: undefined,
+        tokenAddress: undefined,
+      })
+    )
+    this.store.dispatch(this.actions.setRoutes([]))
+
+    this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.LOADING }))
+
+    const { from, to } = this.store.getState().ecoBridge.ui
+
+    if (!from.chainId || !to.chainId || !this._account || !from.address || Number(from.value) === 0) return
+
+    let value = BigNumber.from(0)
+
+    //handling small amounts
+    try {
+      value = parseUnits(from.value, from.decimals)
+    } catch (e) {
+      this.store.dispatch(
+        this.actions.setBridgeDetailsStatus({
+          status: SyncState.FAILED,
+          errorMessage: 'No available routes / details',
+        })
+      )
+      return
+    }
+
+    if (isBridgeSwapActive) {
+      await this._fetchSocketRoutes({
+        fromChainId: from.chainId.toString(),
+        fromTokenAddress:
+          from.address === Currency.getNative(from.chainId).symbol ? SOCKET_NATIVE_TOKEN_ADDRESS : from.address,
+        toTokenAddress: to.address === Currency.getNative(to.chainId).symbol ? SOCKET_NATIVE_TOKEN_ADDRESS : to.address,
+        toChainId: to.chainId.toString(),
+        fromAmount: value.toString(),
+        helperRequestId,
+      })
+    } else {
+      const socketTokens = this.store.getState().ecoBridge.socket.lists[this.bridgeId]
+      const fromNativeCurrency = Currency.getNative(from.chainId)
+
+      let fromTokenAddress = ''
+      let toTokenAddress = ''
+
+      const overrideTokens = overrideTokensAddresses({
+        toChainId: to.chainId,
+        fromChainId: from.chainId,
+        fromAddress: from.address,
+      })
+
+      if (overrideTokens) {
+        const { fromTokenAddressOverride, toTokenAddressOverride } = overrideTokens
+
+        fromTokenAddress = fromTokenAddressOverride
+        toTokenAddress = toTokenAddressOverride
+      }
+
+      // Default pairing
+      if (!toTokenAddress && !fromTokenAddress) {
+        if (from.address === fromNativeCurrency.symbol) {
+          fromTokenAddress = SOCKET_NATIVE_TOKEN_ADDRESS
+          toTokenAddress = SOCKET_NATIVE_TOKEN_ADDRESS
+        } else {
+          const fromToken = socketTokens.tokens.find(
+            token => token.address.toLowerCase() === from.address.toLowerCase()
+          )
+
+          if (!fromToken) {
+            this.store.dispatch(
+              this.actions.setBridgeDetailsStatus({
+                status: SyncState.FAILED,
+                errorMessage: 'No available routes / details',
+              })
+            )
+            return
+          }
+
+          const toToken = socketTokens.tokens.find(
+            token => token.symbol === fromToken.symbol && token.chainId === to.chainId
```

Comment:
Symbol comparsion

Reply:
Okey so on `bridge swap` there is no problem with that because user selects two currencies.

How does it works on normal bridge ?
Socket list looks like:  
```js
    {
      "name": "USD Coin",
      "address": "0x2791bca1f2de4661ed88a30c99a7a9449aa84174",
      "icon": "https://maticnetwork.github.io/polygon-token-assets/assets/usdc.svg",
      "decimals": 6,
      "symbol": "USDC",
      "chainId": "137",
      "logoURI": "https://maticnetwork.github.io/polygon-token-assets/assets/usdc.svg",
      "chainAgnosticId": "DAI"
    }

```
As you can see there is only address on `fromChainId` (in this case on polygon) and now we have to take a look on this part of code:
```js
await QuoteAPI.quoteControllerGetQuote(
        {
          fromChainId,
          fromTokenAddress,
          toTokenAddress,
          toChainId,
          fromAmount,
          userAddress: this._account,
          uniqueRoutesPerBridge: false,
          disableSwapping: false,
          sort: QuoteControllerGetQuoteSortEnum.Output,
          singleTxOnly: true,
        })
```
We need to pass `toTokenAddress` but actually we don't have it on list and to get token address on `toChainId` we did:
```js
const socketListsResponse = await fetch(SOCKET_LISTS_URL)
const socketLists: { data: SocketTokenMap } = await socketListsResponse.json()
this._tokenLists = socketLists.data
```
We are fetching own [token list](https://raw.githubusercontent.com/SwaprDAO/swapr-ecobridge-socket-lists/master/lists/socketList-bidirectional.json)
And inside this [repository](https://github.com/SwaprDAO/swapr-ecobridge-socket-lists) we paired tokens and created map of list.

So yea we are comparing tokens by symbol but:
1. this is only way to do it
2. we have own lists with paired tokens

Reply:
I understand the limitation :)